Common/BaselineSelection.py

```diff
@@ -120,28 +120,52 @@ def GenJetHttOverlapRemoval(df):
 def RequestOnlyResolvedGenJets(df):
     return df.Filter("GenJet_idx[GenJet_B2].size()==2", "Resolved topology")
 
+syst_dict = {
+    "TauES" : ["Tau", "MET"],
+}
+
+def RecoP4(df, syst=None, central_prefix=None):
+    syst_name = syst if syst != None else ''
+    for obj in [ "Electron", "Muon", "Tau", "Jet", "FatJet", "boostedTau","MET", "PuppiMET", "DeepMETResponseTune", "DeepMETResolutionTune"]:
+        if "MET" not in obj:
+            df = df.Define(f"{obj}_idx", f"CreateIndexes({obj}_pt.size())")
+        if syst_name == '':  
+            if "MET" in obj:
+                df = df.Define(f"{obj}_p4", f"LorentzVectorM {obj}_p4({obj}_pt{syst_name}, 0., {obj}_phi{syst_name}, 0.); return {obj}_p4;")
```

Comment:
don't create p4 without suffixes. It will be done in the other method
```python
def SelectRecoP4(df, syst_name):
  for obj in [...]:
    df = df.Define(f"{obj}_p4", f"{obj}_p4_{syst_name}")
  return df
```